PowerPointLabs/PowerPointLabs/ActionFramework/NarrationsLab/TempStorage.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using PowerPointLabs.ActionFramework.Common.Log;
+using PowerPointLabs.TextCollection;
+using PowerPointLabs.Utils;
+using PowerPointLabs.Views;
+
+using Presentation = Microsoft.Office.Interop.PowerPoint.Presentation;
+
+namespace PowerPointLabs.ActionFramework.NarrationsLab
+{
+    public class TempStorage
+    {
+        private const string tempFolderNamePrefix = @"\PowerPointLabs Temp\";
+        private const string SlideXmlSearchPattern = @"slide(\d+)\.xml";
+        private const string fileSuffix = ".pptx";
+        private const string tempZipName = "tempZip.zip";
+
+        private Presentation _currentPresentation;
+        private string _tempPath;
+
+        public string TempPath
+        {
+            get
+            {
+                CheckPresentationChanged();
+                if (_tempPath == null) { throw new NullReferenceException("TempStorage has not been initialized! Call Setup before using."); }
+                return _tempPath;
+            }
+        }
+
+        private void CheckPresentationChanged()
```

Comment:
Consider renaming the function. It does not return a boolean reflecting on the state of the presentation changing. It seems that it performs some actions should the presentation get changed.

Reply:
Can ignore this, narrations lab will be discontinued.